File: recursion/python/Palindrome.py
def is_palindrome(word):
    if (len(word) <= 1):
        return True
    else:
        return word[0] == word[-1] and is_palindrome(word[1:-1])


print(is_palindrome("civic"))
print(is_palindrome("foo"))
print(is_palindrome("racecar"))

File: recursion/typescript/Palindrome.ts


function isPalindrome(word: string): boolean {
  if (word.length <= 1) {
    return true;
  }
  else {
    return word[0] === word[word.length - 1] && isPalindrome(word.slice(1, word.length - 1));
  }
}

console.log(isPalindrome("civic"))
console.log(isPalindrome("foo"))
console.log(isPalindrome("racecar"))
